### PR TITLE
Update the shims

### PR DESCRIPTION
**Pull request by Elv13**
The shims are *not* made to be accurate, but rather to have the minimal/simplest of code required for the higher level code to work.

### CODE REVIEW

**Comment by blueyed** [comment]
I think the diff for this should be covered 100%?!

**Comment by Elv13** [comment]
> I think the diff for this should be covered 100%?!

https://github.com/awesomeWM/awesome/pull/1990 was. Same for all the other PRs I created yesterday. As a whole it had good coverage. But as all the test related commits depends on more than 1 of these PR, I could not push those commits. The whole 30 commit branch has >90% coverage.

File: tests/examples/shims/client.lua [hunk 1]
```diff
@@ -14,6 +14,15 @@ end
 --     CURRENTLY UNUSED
 -- end
 
+local function titlebar_meta(c)
+    for _, position in ipairs {"top", "bottom", "left", "right" } do
+        c["titlebar_"..position] = function(size)--luacheck: no unused
```
Comment (blueyed) [inline tests/examples/shims/client.lua]: Space before `--luacheck`.

File: tests/examples/shims/client.lua [hunk 2]
```diff
@@ -14,6 +14,15 @@ end
...
+            local d = drawin{}
+            return d
```
Comment (blueyed) [inline tests/examples/shims/client.lua]: `return drawin({})` ?

File: tests/examples/shims/client.lua [hunk 3]
```diff
@@ -71,6 +81,20 @@ function client.gen_fake(args)
...
+        return require("beautiful").awesome_icon
+    end
+
+    function ret:raise()
+        --TODO
+    end
+
+    function ret:lower()
+        --TODO
+    end
```
Comment (blueyed) [inline tests/examples/shims/client.lua]: Address the TODOs? / or leave it out for now?!

File: tests/examples/shims/dbus.lua [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        "org.freedesktop.Notifications",
+        data,
+        appname,
+        replaces_id,
+        icon,
+        title,
+        text,
+        actions,
+        hints,
+        expire
```
Comment (blueyed) [inline tests/examples/shims/dbus.lua]: Use `...`, so it becomes `dbus.emit_sigan("org.freedesktop.Notifications", ...)` ?

File: tests/examples/shims/screen.lua [hunk 5]
```diff
@@ -94,8 +98,23 @@ function screen._setup_grid(w, h, rows, args)
...
 
+local function iter_scr(_, _, s)
+    if not s then
+        assert(screen[1])
+        return screen[1], 1
+    end
+
+    local i = s.index
+
+    if i +1  < #screen then
```
Comment (blueyed) [inline tests/examples/shims/screen.lua]: s/+1 /+ 1/

File: tests/examples/shims/client.lua (hunk 3 above)
Comment (Elv13) [inline tests/examples/shims/client.lua]: The functions are called by some of the newer tests, but the behavior is not (yet) required. To keep things simple, I would rather not implement z-index support in this commit.

**Pull request merged**